catalog/spatial/geocoding/spatial-geocoding-websearch/src/main/java/org/codice/ddf/spatial/geocoder/geonames/GeoNamesWebService.java

```diff
@@ -75,28 +106,66 @@ public GeoResult getLocation(String location) {
             LOGGER.error("Error while parsing JSON message from Geonames service.", e);
         }
 
+        return result;
+    }
 
-        if (result != null) {
-            if (result instanceof JSONObject) {
-                JSONObject jsonResult = (JSONObject) result;
-                JSONArray geonames = (JSONArray) jsonResult.get("geonames");
-                if (geonames != null && geonames.size() > 0) {
-                    JSONObject firstResult = (JSONObject) geonames.get(0);
-                    if (firstResult != null) {
-                        double lat = Double.valueOf((String) firstResult.get("lat"));
-                        double lon = Double.valueOf((String) firstResult.get("lng"));
+    WebClient createWebClient(String urlStr) {
+        return WebClient.create(urlStr);
```

Comment:
This could also be done by defining an interface and adding the correct annotations. Then instead of calling "create" each time we could call "geonames.search(location, username)".  This may be useful if we intend to use our own implementation of the GeoNames service and it requires any authentication.